### ISSUE TITLE
embed command that downloads videos from a link and posts it

### ISSUE THREAD

**Issue by Mattwmaster58** [Sooraj-beep/sherp2.0#34]
i dont like going outside discord to watch stuff. esp if they annoy me to login or further distract me from grind grind (reels, shorts, toks, etc)

 i am proposing ?embed command:
 
`?embed <url>` - attempts to download and send that downloaded video in the chat
replying `?embed` to a message with downloadable links - downloads the links.

yt-dlp is the ideal candidate to do the heavy lifting, nearly supports every platform. ffmpeg might be required to remux the videos so they're embedded in discord. 

We can query how large the video is before downloading to see if it goes over the limits so we're not wasting time downloading something we can't send anyway. Bots can send as much as a user, ie 25mb or whatever the server boost level dictates eg 100mb for lvl 3

**Comment by Sooraj-beep** [comment]
I don't think the current free AWS account would be able to handle this

**Comment by Mattwmaster58** [comment]
From a bandwidth or compute perspective?

**Comment by Sooraj-beep** [comment]
Compute

**Comment by DhanrajHira** [comment]
When downloading videos the few times `ffmpeg` is used by `yt-dlp` it is to remux the video. Transcoding is almost never needed so the compute requirements are actually not that high. Sometimes ffmpeg is also used to merge an `hls` stream which once again has very low compute cost.

**Comment by Mattwmaster58** [comment]
Agree. Remuxing is cheap. Some prelimary data I tested [here](https://discord.com/channels/402891511991369740/1139573348406403193) WRT to codec support on desktop:

 - [ ] HEVC
 - [x] h264
 - [ ] h265
 - [x] vp8
 - [x] vp9

_Most_ online services will encode in one of these formats. Youtube for sure and spot checked Instagram, tikTok, (check with yt-dlp --list-formats) as well. So I think in the median case, we only have to remux which is cheap as @DhanrajHira mentioned. And as a fallback, we can transcode if no other options are available. It would need to be implemented thoughtfully tho to prevent a DDoS with all the compute transcode takes. 

Maybe only one transcode job at a time. jumping the gun here a but, but asyncio.Semaphore probably provide an easy solution here.
